wrappers/kn/src/nonAndroidNativeTest/kotlin/BitmapImageView.kt

```diff
@@ -0,0 +1,23 @@
+import korlibs.image.bitmap.Bitmap
+import kotlinx.cinterop.*
+import zxingcpp.ImageFormat
+import zxingcpp.ImageView
+
+class BitmapImageView(
+    val bitmap: Bitmap,
+    override val left: Int = 0,
+    override val top: Int = 0,
+    override val width: Int = bitmap.width,
+    override val height: Int = bitmap.height,
+    override val rotation: Int = 0,
+) : ImageView() {
+    @OptIn(ExperimentalForeignApi::class)
+    override val data: UByteArray
+        get() = bitmap.toBMP32IfRequired().ints.usePinned {
```

Comment:
I understand too little about kotlin/native at the moment to asses how to best model this. One thing is clear though: in case the bitmap is e.g. greyscale, then you should definitively not convert it to RGB first only so that zxing-cpp then internally converts it back again.

Reply:
Will all the images be converted greyscale(aka. Bitmap1)?
If so, I will convert them into bitmap1 in kotlin native by the decoding library.
I do this because I didn't found a good way to decode images easily in kotlin/native. Only to find this api which can get decoded image's original data, there's no public function by what I can get ByteArray.

Actually, I'd say I'm not familiar with native development and image related things at all, the only thing I mastered besides java/kotlin development is to google and read codes and documents

Reply:
Please give this a look, whether do you think is this a appropriate method to decode images?
https://github.com/zxing-cpp/zxing-cpp/pull/719/commits/4904a7e7d86e03831856151133681aa43d4ea847

Reply:
And here is a new one, which directly decode image by stb_image into grey scale: https://github.com/zxing-cpp/zxing-cpp/pull/719/commits/25d7cc8e4c62fb4ae736421ce086276f5e4e4521

Reply:
This stb_image library is a part of(binded by) kirim(A graphic library for k/n) but they didn'enable building for androidNative target.
Although it's easy to create a binding to stb_image, I don't think it would be a good idea.
So now we have two choice:
a) Use the binding form Kirim and suffer from it's huge dependecy chain(no affect on main source set), but no need to have another project wrapping stb_image.
b) Create an kn library for stb_image, we don't need to model it into oop,we can just use it's raw functions, so that we dont need to spend much time on it, and we can set it to target on all the targets. However, this will result in an additional library(in my repository or in zxing-cpp org's repository).